### PR TITLE
Extends rsyslog_logfiles_attributes_modify template for permissions

### PR DESCRIPTION
**Pull request by marcusburghardt**
#### Description:

The PR #9789 introduced a new template to check and remediate rsyslog log files attributes.
The initial template covers `ownership` and `group-ownership` attributes, which satisfy these two rules:

- rsyslog_files_ownership
- rsyslog_files_groupownership

The template is already compatible to both legacy and RainerScript syntax.
This PR is a extension of the `rsyslog_logfiles_attributes_modify` template in order to also cover `permissions` attributes of rsyslog log files, which satisfy the following additional rule:

- rsyslog_files_permissions

#### Rationale:

These are the only three relevant rules for rsyslog log files attributes, which are now aligned to the same approach through this new template. In addition:

- Many test scenario scripts were removed in favor of the templated ones
- The same OVAL, Bash and Ansible logic is used for the three rules
- RainerScript syntax is now supported to find rsyslog log files
- Legacy syntax remains compatible in all rules

#### Review Hints:

- Better to review the PR commit by commit in sequence.
- I recommend automatus to check the rules:

./build_product rhel7 rhel8 rhel9 ...

Example of `automatus` command for testing these 3 rules:
```bash
for rule in rsyslog_files_permissions rsyslog_files_groupownership rsyslog_files_ownership; do
  for flavor in bash ansible; do
    ./tests/automatus.py rule --libvirt qemu:///session <VM NAME> --datastream build/ssg-rhel8-ds.xml --dontclean --remediate-using $flavor $rule
  done
done
```

### CODE REVIEW

**Comment by marcusburghardt** [comment]
FYI @dodys and @teacup-on-rockingchair

**Comment by marcusburghardt** [comment]
`Automatus CS9 / Run Tests (pull_request)` is failing for Bash remediation because there is no `which` command in the container.

**Comment by marcusburghardt** [comment]
> `Automatus CS9 / Run Tests (pull_request)` is failing for Bash remediation because there is no `which` command in the container.

In any case, I fixed this in the last commit.

**Review by yuumasato — CHANGES_REQUESTED** [review]

File: shared/templates/rsyslog_logfiles_attributes_modify/tests/rainer_include_correct_attr.pass.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+{{% if ATTRIBUTE == "owner" %}}
+CHATTR="chown"
+ATTR_VALUE="root"
+{{% elif ATTRIBUTE == "groupowner" %}}
+CHATTR="chgrp"
+ATTR_VALUE="root"
+{{% else %}}
+CHATTR="chmod"
+ATTR_VALUE="0600"
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply shared/templates/rsyslog_logfiles_attributes_modify/tests/rainer_include_correct_attr.pass.sh:15]: In all current cases, the expected permission is `0600` or `0640` and a stricter permission is accepted. So for the test scenarios in `template`, using `0600` is enough for the correct permission. In a similar logic, in all current cases, no permission for "others" is allowed. So `0666` is also sufficient for the test scenarios in this template.

Something interesting would be to include an additional test scenario specific to Ubuntu and SLE in the `rsyslog_files_permissions` rule. I will include.

File: shared/templates/rsyslog_logfiles_attributes_modify/tests/legafy_incorrect_attr.fail.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply shared/templates/rsyslog_logfiles_attributes_modify/tests/legafy_incorrect_attr.fail.sh]: Fixed

File: linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_files_permissions/rule.yml [hunk 3]
```diff
@@ -52,3 +52,9 @@ ocil: |-
     file, run the following command:
     <pre>$ ls -l <i>LOGFILE</i></pre>
     The permissions should be 600, or more restrictive.
+
+template:
+  name: rsyslog_logfiles_attributes_modify
+  vars:
+    attribute: permissions
+    value: '0600'
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply linux_os/guide/system/logging/ensure_rsyslog_log_file_configuration/rsyslog_files_permissions/rule.yml:66]: Verified the respective benchmarks and updated the `rule.yml` file.

File: shared/templates/rsyslog_logfiles_attributes_modify/bash.template [hunk 4]
```diff
@@ -86,10 +86,13 @@ FILE_PARAM="{{{ ATTRIBUTE }}}"
...
      "groupowner")
-        FILE_CMD=$(which chgrp)
+        FILE_CMD="chgrp"
         ;;
      "owner")
-        FILE_CMD=$(which chown)
+        FILE_CMD="chown"
+        ;;
+	 "permissions")
+        FILE_CMD="chmod"
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply shared/templates/rsyslog_logfiles_attributes_modify/bash.template]: Improved the bash template.

File: shared/templates/rsyslog_logfiles_attributes_modify/ansible.template [hunk 5]
```diff
@@ -61,8 +61,9 @@
 - name: '{{{ rule_title }}} -Setup log files attribute'
   ansible.builtin.file:
     path: "{{ item }}"
-    owner: '{{ ( "{{{ ATTRIBUTE }}}" is match("owner")) | ternary({{{ VALUE }}}, omit) }}'
-    group: '{{ ( "{{{ ATTRIBUTE }}}" is match("groupowner")) | ternary({{{ VALUE }}} , omit) }}'
+    owner: '{{ ( "{{{ ATTRIBUTE }}}" is match("owner")) | ternary("{{{ VALUE }}}", omit) }}'
+    group: '{{ ( "{{{ ATTRIBUTE }}}" is match("groupowner")) | ternary("{{{ VALUE }}}" , omit) }}'
+    mode: '{{ ( "{{{ ATTRIBUTE }}}" is match("permissions")) | ternary("{{{ VALUE }}}" , omit) }}'
```
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply shared/templates/rsyslog_logfiles_attributes_modify/ansible.template]: There was issues with this suggestion. I fixed them and amended the commit.

File: shared/templates/rsyslog_logfiles_attributes_modify/tests/rainer_include_correct_attr.pass.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (marcusburghardt) [reply shared/templates/rsyslog_logfiles_attributes_modify/tests/rainer_include_correct_attr.pass.sh:15]: Done.

**Review by yuumasato — APPROVED** [review]
Nice work

**Comment by yuumasato** [comment]
The failure on c9s is unrelated to this PR:
```
:: [ 11:08:46 ] :: [   FAIL   ] :: Rules not passing after remediation:

xccdf_org.ssgproject.content_rule_service_rngd_enabled - fail
```

**Pull request merged**